### PR TITLE
creates secure rich text editor component.

### PR DESCRIPTION
**Pull request by Akshansr**
Secure rich text editor provides a safe and versatile rich text editing experience, with essential formatting tools while restricting URL and image insertions to enhance application security and prevent XSS attacks.

### CODE REVIEW

File: src/components/Pega_Extensions_SecureRichText/index.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        if (!editorValue) {
+          pConn.getValidationApi().validate(editorValue);
+        }
+        if (value !== editorValue) {
+          handleEvent(actionsApi, 'changeNblur', property, editorValue);
+        }
+      }
+    };
+
+    const onImageAdded: EditorProps['onImageAdded'] = (image, id) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_SecureRichText/index.tsx:164]: @Akshansr why do we need this function if you cannot add images?

File: src/components/Pega_Extensions_SecureRichText/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import { EmailDisplay, PhoneDisplay } from '@pega/cosmos-react-core';
+
+type ActionType = (propName: string, value: string) => void;
+
+export const handleEvent = (
```
Comment (ricmars) [inline src/components/Pega_Extensions_SecureRichText/utils.tsx:5]: if the onImageAdded is removed - this function is only used for ChangeNBlur - in this case, we should remove and move the 2 lines of code in the index

File: src/components/Pega_Extensions_SecureRichText/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    case 'Phone': {
+      displayComponent = <PhoneDisplay value={value} variant='link' />;
+      break;
+    }
+    // no default
+  }
+  return displayComponent;
+};
+
+export const updateContentWithAbsoluteURLsOfImgSrcs = (content: string, pConn: typeof PConnect) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_SecureRichText/utils.tsx:59]: Why do we need this if you cannot inject images?

**Review by ricmars — COMMENTED** [review]
There are several issues with the PR
- If you cannot add images or URL, all the handlers should be removed - there are code in utils and index that seems unnecessary
- You can still inject img and url if you open your browser or paste html markup in the value prop - the code should sanetize and remove any img or url before sending it to the server - we need to use DOMPurify to clean up the markup and avoid any injecting if we want to advertise that this component is safer

File: src/components/Pega_Extensions_SecureRichText/index.tsx (hunk 1 above)
Comment (Akshansr) [reply src/components/Pega_Extensions_SecureRichText/index.tsx]: Removed all image handlers @ricmars

File: src/components/Pega_Extensions_SecureRichText/utils.tsx (hunk 3 above)
Comment (Akshansr) [reply src/components/Pega_Extensions_SecureRichText/utils.tsx]: Removed it

File: src/components/Pega_Extensions_SecureRichText/index.tsx (hunk 1 above)
Comment (Akshansr) [reply src/components/Pega_Extensions_SecureRichText/index.tsx]: We added sanitisation to filter out all images and links from content before its sent to server

**Comment by Akshansr** [comment]
> There are several issues with the PR
> 
> * If you cannot add images or URL, all the handlers should be removed - there are code in utils and index that seems unnecessary
> * You can still inject img and url if you open your browser or paste html markup in the value prop - the code should sanetize and remove any img or url before sending it to the server - we need to use DOMPurify to clean up the markup and avoid any injecting if we want to advertise that this component is safer

Now even if user tries to hardcode images/ links in value or if he pastes images / links by mistake, the content will be sanitized first and then inserted into editor.

**Pull request merged**